draft-ietf-quic-transport.md

```diff
@@ -2205,16 +2205,29 @@ connection.
 The endpoint MUST use unpredictable data in every PATH_CHALLENGE frame so that
 it can associate the peer's response with the corresponding PATH_CHALLENGE.
 
+An endpoint MUST expand datagrams that contain a PATH_CHALLENGE frame to at
+least the smallest allowed maximum datagram size of 1200 bytes.  Sending UDP
+datagrams of this size ensures that the network path from the endpoint to the
+peer can be used for QUIC; see {{datagram-size}}.
+
 
 ### Path Validation Responses
 
 On receiving a PATH_CHALLENGE frame, an endpoint MUST respond by echoing the
-data contained in the PATH_CHALLENGE frame in a PATH_RESPONSE frame. A
-PATH_RESPONSE frame does not need to be sent on the network path where the
-PATH_CHALLENGE was received; a PATH_RESPONSE can be sent on any network path.
-An endpoint MUST NOT delay transmission of a packet containing a PATH_RESPONSE
+data contained in the PATH_CHALLENGE frame in a PATH_RESPONSE frame.  An
+endpoint MUST NOT delay transmission of a packet containing a PATH_RESPONSE
 frame unless constrained by congestion control.
 
+A PATH_RESPONSE frame SHOULD be sent on the network path where the
```

Comment:
We can require this from the endpoint, while at the same time requiring the peer to not enforce it. We've done unenforceable MUSTs in other parts of the specs. What is the justification for allowing this wiggle room?

```suggestion
A PATH_RESPONSE frame MUST be sent on the network path where the
```